File: 스택/큐/기능 개발.py
def solution(progresses, speeds):
    answer = []
    day = 0
    count = 0
    # 진행할 작업이 없을 때까지 진행
    while len(progresses) > 0:
        if progresses[0] + day * speeds[0] >= 100:
            # 끝난 count 증가
            count += 1
            # 작업 끝난 애들은 삭제
            progresses.pop(0)
            speeds.pop(0)
        else:
            # 100이 안 넘는데 count가 0보다 큼
            # 즉, 배포해야할 작업이 존재함
            if count > 0: 
                answer.append(count)
                # 다시 으로 초기화
                count = 0
            # day를 100이 안 넘는 경우에만 증가시켜야함 
            # -> 뒤에 기능이 먼저 개발될 수도 있기 때문
            day += 1
        
    answer.append(count)
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
import java.io.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> list = new ArrayList<>();
        
        int i=0;
        while(i<progresses.length){
            if(progresses[i] < 100) {
                int day = (100-progresses[i]) / speeds[i];
                if((100-progresses[i]) % speeds[i] != 0){
                    day += 1;
                }
            
                // 다른거 진행될 동안 진행된 날 수 만큼 다른 작업들도 진행이 되어야함 -> 반영하기
                for(int j=i; j<progresses.length; j++){
                    progresses[j] += (speeds[j] * day);
                }
            }
            
            int count = 0;
            while (i < progresses.length && progresses[i] >= 100) {
        count++;
        i++;
    }
    
    // 3. 이번에 한 번에 배포된 기능의 개수를 리스트에 저장
    list.add(count);
            
        }
        
        int[] answer = new int[list.size()];
        
        for(int a=0; a<list.size(); a++){
            answer[a] = list.get(a);
        }
        
        return answer;
    }
}